chain/chain/src/store.rs

```diff
@@ -2774,7 +2783,7 @@ impl<'a> ChainStoreUpdate<'a> {
         }
         for mut wrapped_trie_changes in self.trie_changes.drain(..) {
             wrapped_trie_changes
-                .wrapped_into(&mut store_update)
+                .wrapped_into(&mut store_update, self.chain_store.archival)
                 .map_err(|err| ErrorKind::Other(err.to_string()))?;
```

Comment:
I don’t quite like the argument here to be honest.  I wonder if it
would be better to inline `wrapped_into`, e.g.:

```rust
            wrapped_trie_changes.insertions_into(store_update)
                .map_err(|err| ErrorKind::Other(err.to_string()))?;
            wrapped_trie_changes.state_changes_into(store_update);
            if !self.chain_store.archival {
                wrapped_trie_changes
                    .trie_changes_into(&mut store_update)
                    .map_err(|err| ErrorKind::Other(err.to_string()))?;
            }
```

Not sure if that’s better.  Especially since sadly try blocks are not
a thing yet and this adds more `map_err`.

By the way, (and this is beyond the scope of this PR of course),
there’s a weird discrepancy in WrappedTrieChanges.  state_changes_into
method cannot fail and it simply asserts that Borsh serialisation
cannot fail.  On the other hand, wrapped_into accepts that set_ser may
fail where the only possible failure for that function is Borsh
serialisation.